main/core/src/eval/Evaluator.scala

```diff
@@ -362,15 +382,302 @@ case class Evaluator(home: os.Path,
       }
     }
 
+    logEnd.foreach(_())
+
     multiLogger.close()
 
     (newResults, newEvaluated)
   }
 
   def resolveLogger(logPath: Option[os.Path], logger: Logger): Logger = logPath match{
     case None => logger
-    case Some(path) => MultiLogger(logger.colored, log, FileLogger(logger.colored, path, debugEnabled = true))
+    case Some(path) => MultiLogger(logger.colored, log, new FileLogger(logger.colored, path, debugEnabled = true))
   }
+
+  type Terminal = Either[Task[_], Labelled[Any]]
+  type TerminalGroup = (Terminal, Agg[Task[_]])
+
```

Comment:
Could we have a short block comment describing the high level algorithm? That will surely come in useful to future maintainers

Reply:
Done